dev/tasks/python-wheels/win-build.bat

```diff
@@ -72,6 +72,9 @@ pushd %ARROW_SRC%\python
 set PYARROW_BUILD_TYPE=Release
 set SETUPTOOLS_SCM_PRETEND_VERSION=%PYARROW_VERSION%
 
+@rem Newer Cython versions are not available on conda-forge
```

Comment:
Can you comment why it's important to get the newer Cython versions?

Reply:
We rely on certain fixes that are part of the latest Cython version and changed our pinning to `>=0.29` some time ago. This then sadly broke the windows wheels build. The main problem here is not actually that we want a newer Cython but that since some time `conda-forge` no longer builds packages for Python 3.5.

Reply:
Then perhaps `pip install Cython` or `pip install -U Cython` (if it's already installed) is better than explicit pinning of a given version?

Reply:
I changed it to `>=0.29` as I don't want to use `-U`.